kvbc/tools/db_editor/include/kv_blockchain_db_editor.hpp

```diff
@@ -703,6 +705,68 @@ struct RemoveMetadata {
   }
 };
 
+struct GetSTMetadata {
+  const bool read_only = true;
+  std::string description() const {
+    return "getSTMetadata\n"
+           "  Shows State Transfer metadata ";
+  }
+
+  std::string to_json(const bftEngine::bcst::impl::DataStore::CheckpointDesc &chckpDesc,
+                      const bftEngine::bcst::impl::DataStore::ResPagesDescriptor *rpDesc) const {
+    std::ostringstream oss;
+    oss << "{\"checkpointNum\": " << chckpDesc.checkpointNum << ", \"lastBlock\": " << chckpDesc.lastBlock
+        << ", \"digestOfLastBlock\": \"" << chckpDesc.digestOfLastBlock.toString()
+        << "\", \"digestOfResPagesDescriptor\": \"" << chckpDesc.digestOfResPagesDescriptor.toString() << "\"";
+    if (rpDesc) {
+      oss << ", \"reserved_pages\": [";
+      for (uint32_t i = 0; i < rpDesc->numOfPages; ++i)
+        if (rpDesc->d[i].relevantCheckpoint > 0) {
+          oss << "{\"pageId\": " << rpDesc->d[i].pageId << ","
+              << "\"relevantCheckpoint\": " << rpDesc->d[i].relevantCheckpoint << ","
+              << "\"pageDigest\": \"" << rpDesc->d[i].pageDigest.toString() << "\"}";
+          if (i < rpDesc->numOfPages - 1) oss << ", ";
+        }
+      oss << "]";
+    }
+    oss << "}";
```

Comment:
Can it be implemented by passing a map to `toJson()`, in order to avoid "manual" formatting? Or was it actually done in that way for better formatting than what `toJson()` can offer?

Maybe we need to change `toJson()` so that it supports nesting better or use a library for it.

Reply:
We have to improve our toJson to recursively support various types of nested containers, I would be more than happy not to implement this formatting by myself...

Reply:
I'll do